output/file/outputfile.go

```diff
@@ -0,0 +1,219 @@
+	DirMode         string `json:"dir_mode"`          // Dir access mode to use. Example: "dir_mode" => 0750
+	FileMode        string `json:"file_mode"`         // File access mode to use. Example: "file_mode" => 0640
+	FlushInterval   int    `json:"flush_interval"`    // Flush interval (in seconds) for flushing writes to log files. 0 will flush on every message.
+	Path            string `json:"path"`              // The path to the file to write. Event fields can be used here, like /var/log/logstash/%{host}/%{application}
+	Codec           string `json:"codec"`             // expression to write to file. E.g. "%{log}"
+	WriteBehavior   string `json:"write_behavior"`    // If append, the file will be opened for appending and each new event will be written at the end of the file. If overwrite, the file will be truncated before writing and only the most recent event will appear in the file.
+	writers         map[string]chan string
+	fileMode        os.FileMode
+	dirMode         os.FileMode
+	fs              fileSystem
+}
+
+// DefaultOutputConfig returns an OutputConfig struct with default values
+func DefaultOutputConfig() OutputConfig {
+	return OutputConfig{
+		OutputConfig: config.OutputConfig{
+			CommonConfig: config.CommonConfig{
+				Type: ModuleName,
+			},
+		},
+		CreateIfDeleted: defaultCreateIfDeleted,
+		DirMode:         defaultDirMode,
+		FileMode:        defaultFileMode,
+		FlushInterval:   defaultFlushInterval,
+		WriteBehavior:   defaultWriteBehavior,
+		Codec:           defaultCodec,
+		writers:         make(map[string]chan string),
+		fs:              osFS{},
+	}
+}
+
+// InitHandler initialize the output plugin
+func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeOutputConfig, error) {
+	conf := DefaultOutputConfig()
+	err := config.ReflectConfig(raw, &conf)
+	if err != nil {
+		return nil, err
+	}
+	if conf.Path == "" {
+		return nil, errors.New("no path defined for file")
+	}
+	if conf.WriteBehavior != "append" && conf.WriteBehavior != "overwrite" {
+		return nil, errors.Errorf("invalid write_behavior defined: %v", conf.WriteBehavior)
+	}
+	var fMode, dMode int
+	fMode, err = strconv.Atoi(conf.FileMode)
+	if err != nil || fMode < 0 {
+		return nil, errors.Errorf("invalid file mode %v: %v", conf.FileMode, err)
+	}
+	conf.fileMode = os.FileMode(fMode)
+	dMode, err = strconv.Atoi(conf.DirMode)
+	if err != nil || dMode < 0 {
+		return nil, errors.Errorf("invalid dir mode %v: %v", conf.DirMode, err)
+	}
+	conf.dirMode = os.FileMode(dMode)
+
+	return &conf, nil
+}
+
+func (t *OutputConfig) createFile(path string) (file, error) {
+	if !t.exists(path) {
+		dir := filepath.Dir(path)
+		if dir != "." && !t.exists(dir) {
+			if err := t.fs.MkdirAll(dir, t.dirMode); err != nil {
+				fmt.Printf("error creating directory: %v\n", t.WriteBehavior)
```

Comment:
There is a logger for error message. [config/goglog/goglog.go](https://github.com/tsaikd/gogstash/blob/master/config/goglog/goglog.go)